### PR TITLE
Example of master/slave update

### LINKED ISSUE #739: Example of inclusive language update
A developer at LLNL was involved in updating a flagship project's inclusive language posture for code, documentation and other materials. I've made initial contact for a possible blog article. They are still in process of deploying the updated code and documentation and will circle back when that is completed.

### PR DESCRIPTION
**Pull request by markcmiller86**
# Description

EB Member: @rinkug and @curfman 

Resolves #739

## PR checklist for files displayed on bssw.io site

* [x] ***[Author]*** `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* [x] ***[Author]*** Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py])~~.
* [x] ***[Author]*** Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [x] ***[EB Mem]*** Assign this PR to the EB member `<eb-member-id>`.
* [x] ***[EB Mem]*** Assign this PR to the author of the PR `<pr-author-id>`.
* [x] ***[EB Mem]*** Add one or more Reviewers.
* [x] ***[EB Mem]*** Add label `content: <content-type>` for the type of contribution.
* [x] ***[EB Mem]*** Add to Project `Content Development` (see [Content Development]).
* [x] ***[EB Mem]*** Add [meta-data] to the `*.md` file(s) (set `Publish: yes`).
* [x] ***[EB Mem]*** Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
* [x] ***[EB Mem]*** Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [x] ***[Author]*** Make any final changes to the PR based on feedback.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py]).~~
* [x] ***[EB Mem]*** Rebuild [preview] site and re-confirm content looks correct.
* [x] ***[EB Mem]*** Ensure at least one reviewer signs off on the final changes.
* [x] ***[EB Mem]*** Move the PR to "Ready to Publish" in [Content Development].
* [x] ***[EB Mem]*** Leave comment and @mention `@betterscientificsoftware/bssw-maint` (**BSSw Maint**) asking to carry out final publication steps.
* [ ] ***[BSSw Maint]*** Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [ ] ***[BSSw Maint]*** Merge PR. (Should automatically move to "Done" in [Content Development].)
* [ ] ***[BSSw Maint]*** Verify new contribution shows up on [bssw.io] as expected.

NOTE:
* Checklist items prefixed with ***[Author]*** are expected to be performed by the author of the PR or can be performed by the author.
* Checklist items prefixed with ***[EB Mem]*** must be performed by a [BSSw.io Editorial Board (EB) Member](https://betterscientificsoftware.github.io/bssw.io/bssw_members.html).
* Checklist items prefixed with ***[BSSw Maint]*** must be performed by someone in the `@betterscientificsoftware/bssw-maint` team (hint: type `@`, `b`, `s`, `s`, `w`, `-`, `m`  to auto-complete to `@betterscientificsoftware/bssw-maint`).

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

**Comment by markcmiller86** [comment]
Possible images to include in banner image

Don't know how to request permission on this one...

image

Am seeking permission on this one...

image

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+One way software projects address inclusion is to identify and improve language used in their projects representing barriers to inclusion.
+An unfortunately [prolific](https://github.com/search?q=master+slave&type=code) and unmistakably [problematic](https://www.wired.com/story/tech-confronts-use-labels-master-slave/) example in HPC/CSE projects is the use of `master/slave` terminology.
+
+The terminology may have [taken root](https://www.cise.ufl.edu/~sahni/papers/masterslave2.pdf) in technology in the 1940s.
+Examination of [Google's NGram viewer](https://books.google.com/ngrams/graph?content=master-slave&year_start=1800&year_end=2019&case_insensitive=on&corpus=26&smoothing=3&direct_url=t4%3B%2Cmaster%20-%20slave%3B%2Cc0%3B%2Cs0%3B%3Bmaster%20-%20slave%3B%2Cc0%3B%3BMaster%20-%20Slave%3B%2Cc0%3B%3BMaster%20-%20slave%3B%2Cc0%3B%3BMASTER%20-%20SLAVE%3B%2Cc0) shows a sharp rise in its use beginning with [Raymond Geortz's](https://en.wikipedia.org/wiki/Raymond_Goertz) description of a [remote manipulator](https://en.wikipedia.org/wiki/Remote_manipulator) which is a sort of robotic version of a [glove box](https://en.wikipedia.org/wiki/Glovebox).
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:7]:
```suggestion
Examination of [Google's NGram viewer](https://books.google.com/ngrams/graph?content=master-slave&year_start=1800&year_end=2019&case_insensitive=on&corpus=26&smoothing=3&direct_url=t4%3B%2Cmaster%20-%20slave%3B%2Cc0%3B%2Cs0%3B%3Bmaster%20-%20slave%3B%2Cc0%3B%3BMaster%20-%20Slave%3B%2Cc0%3B%3BMaster%20-%20slave%3B%2Cc0%3B%3BMASTER%20-%20SLAVE%3B%2Cc0) shows a sharp rise in its use beginning with [Raymond Geortz's](https://en.wikipedia.org/wiki/Raymond_Goertz) description of a [remote manipulator](https://en.wikipedia.org/wiki/Remote_manipulator), which is a sort of robotic version of a [glove box](https://en.wikipedia.org/wiki/Glovebox).
```

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+The work was spearheaded by [Ben Liu](https://scholar.google.com/citations?user=pA2ybRwAAAAJ&hl=en) of the ALE3D team and [Gabriel Jose de Frias](https://www.osti.gov/search/author:%22de%20Frias,%20Gabriel%20Jose%22) of the Sierra team.
+
+If the work had involved only a global, textual replacement of the words `master` and `slave`, there would not have been much reason for writing about it here.
+The work involved changes to source code, dependency interfaces, documentation, test input parameter files and test data files as well as how developers and users alike talk about features of the code where this terminology was previously used.
+
+Managers within both projects had a long-standing interest in replacing `master/slave` terminology.
+However, they recognized, wisely, that such activities are far more likely to succeed when they happen organically rather than by management mandate.
+Finding a champion as well as mustering the consensus and resources to take on the work remained elusive.
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:29]:
```suggestion
Finding a champion, as well as mustering the consensus and resources to take on the work, remained elusive.
```

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+This action was intended to encourage collaborators to consider replacing the terminology and to broaden consensus on common replacement terms.
+
+Within the [NNSA labs](https://www.energy.gov/nnsa/national-nuclear-security-administration), the terms `primary` and `secondary` are commonly associated with the major components of a [thermonuclear weapon](https://cgsr.llnl.gov/content/assets/docs/CGSR_NW101_Policy_Wonks_11-04-21_WEB_v5.pdf?#page=19).
+Their presence is often accompanied by classified information handling requirements.
+Although the terms `primary` and `secondary` are not themselves classified, using them as prolifically as `master` and `slave` have been used would present significant challenges in ensuring clean and clear separation between classified and unclassified information.
+
+In addition, to be thorough Ben felt it was important to replace not only `master` and `slave` but also any instances of terminology that *derived* from them.
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:51]:
```suggestion
In addition, to be thorough, Ben felt it was important to replace not only `master` and `slave`, but also any instances of terminology that *derived* from them.
```

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+   > After discussions with numerous stakeholders, everyone agreed a suitable replacement was simply *Side A* and *Side B*.
+   > That is somewhat specific to slide surfaces but nonetheless worked for all our cases.
+ 
+2. *Did your efforts exploit automated tooling or workflows to complete the changes?
+   Was it all sort of manual or maybe `grep` and `sed` scripts?*
+
+   > We used `sed` scripts to help update the code and the manual.
+   > We didn't use any compiler tools other than attempting a compile, failing, and fixing it (for the times when the scripts were incorrect).
+   > An extensive pull request was used to identify additional (mostly comment and variable naming) changes.
+   > A number of other developers on the team were very thorough in looking through this
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:67]:
```suggestion
   > A number of other developers on the team were very thorough in looking through this.
```

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+ 
+7. *What role did the social climate in the country in the middle of 2020 play?*
+ 
+   > I think that it brought more awareness (to a broader segment of the population) to how systems and language can reinforce and propagate injustice and oppression.
+   > In particular, how seemingly little things can add up (microaggressions, etc).
+   > For myself, this really made me more aware of the need to change this.
+   > I think this was also true for other developers and users.
+   > I think the broader social climate helped provide momentum to push through the tedium and address resistance to change.
+   > In particular, knowing that Sandia and Idaho National Lab were also working on this helped give me the confidence I needed to continue in what I was doing.
+   > Our work on replacing `master/slave` doesn't necessarily "moved the needle" much.
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:96]:
```suggestion
   > Our work on replacing `master/slave` doesn't necessarily "move the needle" much.
```

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+   > The actual coding time needed to change the code and documentation wasn't the main concern.
+   > The challenge was in having the dialog essential to bring all stakeholders on board.
+ 
+### Gabriel Jose de Frias and Sierra
+
+For Gabriel on Sierra, a decisive moment came after reading an online article about `master/slave` and approaching Sierra project leadership about addressing it in their code base.
+
+1. *Where was the `master/slave` terminology used in the code?*
+
+   > Contact surfaces (e.g. slide surfaces) is the main capability where the terminology was used in Sierra but it was also used in multi-point constraints, periodic boundary conditions and shell-solid joint code.
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:117]:
```suggestion
   > Contact surfaces (e.g. slide surfaces) were the main capability where the terminology was used in Sierra, but it was also used in multi-point constraints, periodic boundary conditions, and shell-solid joint code.
```

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+   > Through sprint reviews every 3 weeks.
+   > We involved analysts and developers in ongoing conversations.
+ 
+7. *What role did the social climate in the country in the middle of 2020 play?*
+
+   > The protests made me think more about what I’m doing in my day-to-day to become a better member of society.
+   > Am I educating myself enough about these issues?
+   > Am I putting myself in other people’s shoes?
+   > Am I considering my own biases in my opinions and thoughts?
+   > I came across the Washington post opinion piece trying to answer some of these questions...trying to get a different perspective from my own, and those around me.
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:150]:
```suggestion
   > I came across the _Washington Post_ opinion piece trying to answer some of these questions...trying to get a different perspective from my own, and those around me.
```

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+
+9. *Did you have a good understanding of what was involved going in?*
+
+   > Agreeing on the replacement terminology certainly took longer than I expected.
+   > That ended up being a good thing since it prompted us to open up the conversation with several teams at Sandia and even other DOE labs.
+
+### Closing thoughts
+
+While there was [much discussion](https://www.google.com/search?q=%22master%22+%22slave%22+terminology+in+software&sxsrf=AOaemvIy5OkrC1xJEV_CJrUNIG3Tvmtlrg:1643308409486&source=lnt&tbs=qdr:y&sa=X&ved=2ahUKEwjdlYPwyNL1AhVuD0QIHXqoBOsQpwV6BAgBEBk&biw=1616&bih=948&dpr=1) of the `master/slave` issue in the media throughout 2020, it has been a [known](https://tools.ietf.org/id/draft-knodel-terminology-00.html) and [long-standing](https://www.jstor.org/stable/40061475) issue long before 2020 with the earliest online mentions going back [nearly 20 years](http://www.cnn.com/2003/TECH/ptech/11/26/master.term.reut/index.html) and other public discourse almost certainly much earlier.
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:165]:
```suggestion
While there was [much discussion](https://www.google.com/search?q=%22master%22+%22slave%22+terminology+in+software&sxsrf=AOaemvIy5OkrC1xJEV_CJrUNIG3Tvmtlrg:1643308409486&source=lnt&tbs=qdr:y&sa=X&ved=2ahUKEwjdlYPwyNL1AhVuD0QIHXqoBOsQpwV6BAgBEBk&biw=1616&bih=948&dpr=1) of the `master/slave` issue in the media throughout 2020, it has been a [known](https://tools.ietf.org/id/draft-knodel-terminology-00.html) and [long-standing](https://www.jstor.org/stable/40061475) issue long before 2020, with the earliest online mentions going back [nearly 20 years](http://www.cnn.com/2003/TECH/ptech/11/26/master.term.reut/index.html), and other public discourse almost certainly much earlier.
```

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+
+Beginning in 2021 less than a year after the summer of 2020, the broader US public has seen what can only be described as a [*backlash*](https://fivethirtyeight.com/features/white-backlash-is-a-type-of-racial-reckoning-too/) against these and other [DE&I efforts](https://sloanreview.mit.edu/article/fighting-backlash-to-racial-equity-efforts/).
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:167]:
```suggestion
Beginning in 2021, less than a year after the summer of 2020, the broader US public has seen what can only be described as a [*backlash*](https://fivethirtyeight.com/features/white-backlash-is-a-type-of-racial-reckoning-too/) against these and other [DE&I efforts](https://sloanreview.mit.edu/article/fighting-backlash-to-racial-equity-efforts/).
```

**Comment by karbarz** [comment]
Finished all of my edits! Great article, Mark!

**Comment by markcmiller86** [comment]
> Finished all of my edits! Great article, Mark!

Thanks @karbarz and thanks for reviewing and corrections 🎉.

For that wapo article, it occurred to me to go find that article. I think its [this one](https://www.washingtonpost.com/opinions/2020/06/12/tech-industry-has-an-ugly-master-slave-problem/) but I am waiting for confirmation from Gabriel on that.

**Comment by markcmiller86** [comment]
Ok, I've inspected preview and noticing numbered lists are not working there...at least not the way I've entered them in this markdown file. We can switch to bullets (easiest) or adjust formatting to conform to what the backend needs for numbered lists.

@sbxchicago do you have any guidance for numbered list handling here?

**Comment by bernhold** [comment]
@markcmiller86 what did you get when you previewed the numbered markup?  Was it setting the list items in paragraph mode?  Or something else?

If I had to *guess*, without doing experiments, the complexity of the content between the list items may be messing up the parser.  As you probably know, Markdown is not that well defined, and various parsers often handle details differently.  If that's the case, I think bullets might be the best alternative.  To me, it looks fine that way.

**Comment by markcmiller86** [comment]
> @markcmiller86 what did you get when you previewed the numbered markup?

I just got all `1`'s. It could have been line breaks between quote blocks and questions. I didn't investigate.

**Comment by markcmiller86** [comment]
@karbarz, @bernhold, @curfman and @rinkug ...

In my mind's eye, I was targeting this for a blog post and so wound up formatting it as such (hero image, etc.) but also with `deck text` (maybe that was wrong?).

Anyhow, I had a thought at the last moment for the *banner* image but it might be somewhat controversial so would appreciate your eye.  Have a look. Its fine if you wanna remove it or maybe delay final pub. to make time for consideration.

**Comment by bernhold** [comment]
> > @markcmiller86 what did you get when you previewed the numbered markup?
> 
> I just got all `1`'s. It could have been line breaks between quote blocks and questions. I didn't investigate.

Yeah, that would be consistent with the content between the list items confusing the parser.  I don't know exactly what.  I actually suspect that whatever GH uses is more generous than the nominal specification for Markdown.

**Comment by karbarz** [comment]
> @karbarz, @bernhold, @curfman and @rinkug ...
> 
> In my mind's eye, I was targeting this for a blog post and so wound up formatting it as such (hero image, etc.) but also with `deck text` (maybe that was wrong?).
> 

@markcmiller86 
I would agree that although this was originally thought of as a CC that it would work better as a blog post. If we chose to take it in that direction, we could save it for next month's digest and choose an older CC to highlight - maybe the inclusive language resources again?

**Comment by markcmiller86** [comment]
Closing pending further LLNL review

**Comment by markcmiller86** [comment]
@rinkug, @bernhold and @curfman and @karbarz this is passed LLNL review and ready for publication.

**Comment by markcmiller86** [comment]
@betterscientificsoftware/bssw-maint ready for publication

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+Yes, they were.
+Hopefully, the community will recognize the importance of unburdening our BIPOC colleagues from having to always be the ones doing the heavy lifting in efforts aimed at making HPC/CSE more inclusive.
+On the other hand, we must also take care that when we engage in such work, we do so with a commitment of ensuring the voices of our BIPOC colleagues are informing priorities and process.
+
+Given the current social justice climate in which we all operate, some readers may feel that inclusive language efforts are nowhere near enough to meet the moment and are really just a [distraction](https://www.wired.com/story/tech-confronts-use-labels-master-slave/) from bigger issues.
+It's hard to argue with that.
+Nonetheless, others may feel like such efforts go way too far.
+While we can acknowledge both perspectives exist, it is worth considering that when we’re [accustomed](https://www.huffpost.com/entry/when-youre-accustomed-to-privilege_b_9460662) to the status quo, being called to invest effort to be more inclusive may feel like oppression.
+
+*Prepared by LLNL under Contract DE-AC52-07NA27344 as document LLNL-AR-832434.*
```
Comment (bernhold) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:188]: Does this need to be visible?  Could we turn it into a comment?

**Review by bernhold — APPROVED** [review]
@markcmiller86 I like the article very much.  Thanks for your work on this!

I'm supposed to be getting input from SBX on the hero.

@karbarz I see Gail tagged as a reviewer.  Do you know whether anyone has actually reached out to her to ask her to edit?  I don't expect she's paying much attention to GH notifications.

**Comment by karbarz** [comment]
@bernhold I'm unsure if anyone's reached out to Gail... @rinkug @curfman?

**Comment by bernhold** [comment]
> @bernhold I'm unsure if anyone's reached out to Gail... @rinkug @curfman?

I don't expect they have.  How about if you go ahead and make an edit pass?

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md (hunk 10 above)
Comment (markcmiller86) [reply Articles/Blog/InclusiveLanguageCodeUpdateExample.md:188]: > Does this need to be visible? Could we turn it into a comment?

Yes, it does. Fortunately, its only a single line 😉

Comment (bernhold) [reply Articles/Blog/InclusiveLanguageCodeUpdateExample.md:188]: Okay.  I realize that we also need your bio. ~50-100 words, under a L4 heading.

**Comment by markcmiller86** [comment]
> @markcmiller86 I like the article very much. Thanks for your work on this!

Thanks for your praise 😃 

> 
> I'm supposed to be getting input from SBX on the hero.

One thing that is very important IMHO about the current image is that it some screen shots of *real code* from some random GitHub project(s) (I poked around some of the 400,000+ repos my search hit to see if there might be verbiage of this ilk there) and it shows just how badly poorly chosen terminology can proliferate, get combined with other terminology, and really look quite bad. Whatever we do settle on for the hero image, I'd like to try to not lose those aspects of the current draft.

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 11]
```diff
@@ -0,0 +1,195 @@
+# Experiences replacing `master/slave` terminology in ALE3D and Sierra
+
+**Hero Image:**
+
+- <img src='../../images/github_ms_snipits.png'/>
+
+#### Contributed by [Mark C. Miller](http://github.com/markcmiller86 "Mark C. Miller")
+#### Publication date: Feb 24, 2022
+
+How do you think code such as depicted in the snippets here, from a short search (<10 min) of GitHub, might impact potential users, developers, collaborators or sponsors?
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:10]:
```suggestion
How do you think such code, as pulled from a short (<10 min) GitHub search and depicted in the snippets above, might impact potential users, developers, collaborators or sponsors?
```

File: Articles/Blog/InclusiveLanguageCodeUpdateExample.md [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+
+* *Did you have a good understanding of what was involved going in?*
+
+   > Agreeing on the replacement terminology certainly took longer than I expected.
+   > That ended up being a good thing since it prompted us to open up the conversation with several teams at Sandia and even other DOE labs.
+
+### Closing thoughts
+
+While there was [much discussion](https://www.google.com/search?q=%22master%22+%22slave%22+terminology+in+software&sxsrf=AOaemvIy5OkrC1xJEV_CJrUNIG3Tvmtlrg:1643308409486&source=lnt&tbs=qdr:y&sa=X&ved=2ahUKEwjdlYPwyNL1AhVuD0QIHXqoBOsQpwV6BAgBEBk&biw=1616&bih=948&dpr=1) of the `master/slave` issue in the media throughout 2020, it has been a [known](https://tools.ietf.org/id/draft-knodel-terminology-00.html) and [long-standing](https://www.jstor.org/stable/40061475) issue long before 2020, with the earliest online mentions going back [nearly 20 years](http://www.cnn.com/2003/TECH/ptech/11/26/master.term.reut/index.html), and other public discourse almost certainly much earlier.
```
Comment (karbarz) [inline Articles/Blog/InclusiveLanguageCodeUpdateExample.md:173]:
```suggestion
While there was [much discussion](https://www.google.com/search?q=%22master%22+%22slave%22+terminology+in+software&sxsrf=AOaemvIy5OkrC1xJEV_CJrUNIG3Tvmtlrg:1643308409486&source=lnt&tbs=qdr:y&sa=X&ved=2ahUKEwjdlYPwyNL1AhVuD0QIHXqoBOsQpwV6BAgBEBk&biw=1616&bih=948&dpr=1) of the `master/slave` issue in the media throughout 2020, it has been a [known](https://tools.ietf.org/id/draft-knodel-terminology-00.html) and [long-standing](https://www.jstor.org/stable/40061475) problem, touched on far before 2020, with the earliest online mentions going back [nearly 20 years],(http://www.cnn.com/2003/TECH/ptech/11/26/master.term.reut/index.html), and other public discourse almost certainly much earlier.
```

**Comment by bernhold** [comment]
> > I'm supposed to be getting input from SBX on the hero.
> 
> One thing that is very important IMHO about the current image is that it some screen shots of _real code_

Understood.  I'm concerned about the appearance when we use it in the background of the title in the blog listing.  To me, its not great.  I'm hoping Matt can play with the colors to make it better.  Fallback is to put that image in the body and probably not use a hero per se.

**Comment by bernhold** [comment]
@markcmiller86 how did you produce that hero image?  Do you have ppt or something that we can send to SBX so that he could more easily manipulate the colors if needed?  Maybe email to me if you do?  Thanks

**Comment by markcmiller86** [comment]
> @markcmiller86 how did you produce that hero image? Do you have ppt or something that we can send to SBX so that he could more easily manipulate the colors if needed? Maybe email to me if you do? Thanks

I can't recall now, that said, I did just [screen capture from a GitHub search](https://github.com/search?q=slave+kill&type=code) some cases that I found (randomly) that seemed to illustrate a key point of the article. At some point, I think I used macOS `preview` and added the colored boxes (color sampling a color from bssw.io site). But, I don't have any of those intermediate steps. Just the final image.

**Pull request merged**